src/steam_network/protocol/protobuf_client.py

```diff
@@ -168,77 +321,211 @@ async def _get_obfuscated_private_ip(self) -> int:
+            if ( logon.Body.obfuscated_private_ip.ShouldSerializev4() )
+            {
+                logon.Body.deprecated_obfustucated_private_ip = logon.Body.obfuscated_private_ip.v4;
+            }
+
+            logon.ProtoHeader.client_sessionid = 0;
+            logon.ProtoHeader.steamid = steamId.ConvertToUInt64();
+
+            logon.Body.account_name = details.Username; //Null
+            logon.Body.password = details.Password; //Null
+            logon.Body.should_remember_password = details.ShouldRememberPassword; //false
+
+            logon.Body.protocol_version = MsgClientLogon.CurrentProtocol; 
+            logon.Body.client_os_type = ( uint )details.ClientOSType; //
+            logon.Body.client_language = details.ClientLanguage;
+            logon.Body.cell_id = details.CellID ?? Client.Configuration.CellID //
+
+            logon.Body.steam2_ticket_request = details.RequestSteam2Ticket;
+
+            // we're now using the latest steamclient package version, this is required to get a proper sentry file for steam guard
+            logon.Body.client_package_version = 1771; // todo: determine if this is still required
+            logon.Body.supports_rate_limit_response = true;
+            logon.Body.machine_name = details.MachineName;
+            logon.Body.machine_id = HardwareUtils.GetMachineID( Client.Configuration.MachineInfoProvider );
+
+            // steam guard 
+            logon.Body.auth_code = details.AuthCode;
+            logon.Body.two_factor_code = details.TwoFactorCode;
+
+#pragma warning disable CS0618 // LoginKey is obsolete
+            logon.Body.login_key = details.LoginKey;
+#pragma warning restore CS0618 // LoginKey is obsolete
+
+            logon.Body.access_token = details.AccessToken;
+
+            logon.Body.sha_sentryfile = details.SentryFileHash;
+            logon.Body.eresult_sentryfile = ( int )( details.SentryFileHash != null ? EResult.OK : EResult.FileNotFound );
+
+
+            this.Client.Send( logon );
+        }
+        """
+        resetSteamIDAfterThisCall : bool = False
+        if (self.confirmed_steam_id is None):
+            resetSteamIDAfterThisCall = True
+            self.confirmed_steam_id = steam_id
+
+        message = CMsgClientLogon()
+        message.client_supplied_steam_id = steam_id
+        message.protocol_version = self._MSG_PROTOCOL_VERSION
+        message.client_package_version = self._MSG_CLIENT_PACKAGE_VERSION
+        message.cell_id = cell_id
+        message.client_language = "english"
+        message.client_os_type = os_value if os_value >= 0 else 0
+        message.obfuscated_private_ip.v4 = await self._get_obfuscated_private_ip()
+        message.qos_level = 3
+        message.machine_id = machine_id
+        message.account_name = account_name
+        #message.password = ""
+        #message.password = None
+        message.should_remember_password = True
+        message.eresult_sentryfile = EResult.FileNotFound
```

Comment:
The new auth flow doesn't use the sentry for steam guard, and iirc it doesn't give it to us, either. I'll need to check. It does provide a "guard data" field but afaik it's not necessary.

The weird part was i used to save the steam guard data the login message returns, but trying to use it would actually break if the user had mobile confirmation for their 2FA. Mobile Confirm 2FA does not use it and it's left blank, and getting to provide it when we logged back in would break.